networking/external_dns_operator/nw-control-dns-records-public-hosted-zone-azure.adoc

```diff
@@ -0,0 +1,122 @@
+:_content-type: PROCEDURE
+[id="nw-control-dns-records-public-hosted-zone-azure_{context}"]
+= Creating DNS records on an public hosted zone for Azure by using Red Hat External DNS Operator
+
+.Procedure
+
+. Export the `kubeconfig` file:
++
+[source,terminal]
+----
+$ export KUBECONFIG=~/<path to>/kubeconfig
+----
+
+. Check the user. The user must have access to the `kube-system` namespace if you don’t have the credentials as you can fetch the credentials from the `kube-system` namespace to use the cloud provider client:
++
+[source,terminal]
+----
+$ oc whoami
+----
++
+.Example output
+[source,terminal]
+----
+system:admin
+----
+
+. Fetch the values from azure-credentials secret present in `kube-system` namespace.
++
+[source,terminal]
+----
+$ CLIENT_ID=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_id}} | base64 -d)
+$ CLIENT_SECRET=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_secret}} | base64 -d)
+$ RESOURCE_GROUP=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_resourcegroup}} | base64 -d)
+$ SUBSCRIPTION_ID=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_subscription_id}} | base64 -d)
+$ TENANT_ID=$(oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_tenant_id}} | base64 -d)
+----
+
+. Login to azure with base64 decoded values:
++
+[source,terminal]
+----
+$ az login --service-principal -u "${CLIENT_ID}" -p "${CLIENT_SECRET}" --tenant "${TENANT_ID}"
+----
+
+. Get the routes to check the domain:
++
+[source,terminal]
+----
+$ oc get routes --all-namespaces | grep console
+----
++
+.Example output
+[source,terminal]
+----
+openshift-console          console             console-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.test-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+----
+
+. Get the dns zone list with respect to your resource group:
++
+[source,terminal]
+----
+$ az network dns zone list --resource-group "${RESOURCE_GROUP}"
+----
+
+. To create `ExternalDNS` CR for `route`, run the following command:
++
+[source,yaml]
+----
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-azure <1>
+spec:
+  domains:
+  - filterType: Include <2>
+    matchType: Exact <3>
```

Comment:
Do we need to specify `name` for `filterType` and `matchType` and to make `domain` part complete?
If no `name` here, maybe above three lines are not required.

cc @miheer

Reply:
Addressed. Dropped the `domain` field